learner/sampler/slurper.py

```diff
@@ -0,0 +1,252 @@
+                    ##If not clean, then jump to the next line by using the \n as a marker.
+                    if eol_idx != -1:
+                        i += eol_idx + 1
+                        break
+                    if i == 0:
+                        break
+                f.seek(i, 0)
+                random_sample.append(f.readline().strip())
+        return random_sample
+    
+    def read_random_xml(file: str = None, sample_size: int = None, byte_bite: int = 50, unit: str = None, encoding: str = None) -> List[bytes]:
+        """Randomly point reader through the file, using the brackets as newline. Iterate until full line."""        
+        if not file:
+            file = self.filename
+        if not sample_size:
+            sample_size = self.sample_size
+        if not unit:
+            unit = self.unit              
+        if not encoding:
+            encoding = self.encoding            
+        
+        start_seq = bytearray("<"+unit, encoding = encoding)
+        transform = bytearray("</"+unit+">", encoding = encoding)
+        start_len = len(start_seq) - 1 
+        tran_len = len(transform) - 1 
+        assert byte_bite > 0, "Search step size must be positive."
+        
+        random_sample = []
+        with open(file, "rb") as f:
+            for sample in range(sample_size):
+                f.seek(0, 2)
+                size = f.tell()
+                i = random.randrange(0, size)
+                while True:
+                    i -= byte_bite
+                    #Prevent an improper seek before (0,0).
+                    if i < 0:
+                        byte_bite += i
+                        i = 0
+                    f.seek(i, 0)
+                    nxt = f.read(byte_bite)
+                    eol_idx = nxt.rfind(transform)
+                    #If not clean, then jump to the next line by using the closing brackets as a marker.
+                    if eol_idx != -1:
+                        i += eol_idx + 1
+                        break
+                    if i == 0:
+                        break
+
+                #This marks the start of the xml data that we want.
+                start_block = i + tran_len
+                f.seek(start_block, 0)
+
+                while True:
+                    i += byte_bite
+                    #Just read until eof if we rolled the very last.
+                    if i > size:
+                        i = f.seek(0,2)
+                        break
+                    f.seek(i, 0)
+                    search = f.read(byte_bite)
+                    start_idx = search.find(start_seq)
+                    if start_idx != -1:
+                        i += start_idx + 1
+                        break
+                    if i == 0:
+                        break
+                f.seek(start_block, 0)
+
+                random_sample.append(f.read(i - start_block - 1))
+            f.close()
+        return random_sample
```

Comment:
Again, this function is huge. Aim to make your functions under 30 lines if possible.  Any longer, and you can probably break it up into functions which do their own thing.